search-sentiments-map/public/scripts/modal-script.js

```diff
@@ -154,59 +188,81 @@ function highlightWords(
   let startDisplayingSnippetIndex = 0;
   wordsToHighlight.forEach((word) => {
+    // The sentiment node.js module always returns the positive or negative 
+    // words in lowercase, so make only the snippet lowercase.
     let wordIndex = snippet.toLowerCase().indexOf(word);
     if (wordIndex !== -1){
-      let positiveHighlight = '<span style="color: ' +
-          POSITIVE_COLOR + ';">' +
-          snippet.substring(wordIndex, wordIndex + word.length) +
-          '</span>';
-      let negativeHighlight = '<span style="color: ' +
-          NEGATIVE_COLOR + ';">' +
-          snippet.substring(wordIndex,  wordIndex + word.length) +
+      let highlight = '<span style="color: ' +
+          (scoreIsPositive ? POSITIVE_COLOR : NEGATIVE_COLOR) +
+          ';">' + snippet.substring(wordIndex, wordIndex + word.length) +
           '</span>';
-      snippet = scoreIsPositive ?
-          snippet.replace(new RegExp( '\\b'+ word + '\\b', 'gi'), positiveHighlight) : 
-          snippet.replace(new RegExp('\\b' + word + '\\b', 'gi'), negativeHighlight);
+      snippet = snippet.replace(new RegExp('\\b' + word + '\\b', 'gi'), highlight);
     }
   });
   resultElement.innerHTML += snippet;
 }
 
 /** 
- * Displays the popularity timeline for the current search topic in the 
+ * Sets the popularity timeline for the current search topic in the 
  * selected country's modal.
- * @param {string} countryCode The two-letter code of the selected country.
  */
-function displayPopularityTimeline(countryCode) {
+function setPopularityTimeline() {
   const popularityTimelineElement = document.getElementById('popularity-timeline-tab');
   popularityTimelineElement.innerHTML = '';
-  fetch('/trends/', {
+  postTrendsToGetPopularityTimelineData(getCurrentSearchData().topic, countryCode)
+      .then(timelineJSON => {
+        if (timelineJSON.default.timelineData.length === 0) {
+          popularityTimelineElement.innerText = 
+              'Popularity Timeline is not available for the selected country.';
+        } else {
+          drawPopularityTimeline(
+              timelineJSON.default.timelineData, popularityTimelineElement, 
+              getCurrentSearchData().topic);
+       }
+  });
+}	
+
+/** 
+ * Gets the popularity timeline data from the Google Trends API if it is not
+ * stored in the local cache. Prevents making too many calls to the API if the 
+ * user continuously resizes the window.
+ * @param {string} topic The search topic that the popularity timeline is based on.
+ * @param {string} countryCode Two letter country code for selected country.
+ * @param {Object} popularityTimelineElement The element where the popularity 
+ * timeline is updated.
+ * @return {Promise} The promise which resolves to the the timeline JSON data.
+ */
+function postTrendsToGetPopularityTimelineData(
+    topic, countryCode, popularityTimelineElement) {
+  // Check if country code and topic is in the cache already.
+  if (cachePopularityTimelineData[topic]) {
```

Comment:
Aren't you only checking if the topic is in the cache? What about if the topic is in the cache(because you just added it for a different country, but this country does not exist yet?